### ISSUE TITLE
Question about alerts to a Slack channel: redirect URL and Hard Coded Info

### ISSUE THREAD

**Issue by DeaMariaLeon** [conbench/conbench#1574]
I can send conbench alerts to a Slack channel in my workspace:
<img width="450" alt="Screenshot 2024-01-16 at 18 58 08" src="https://github.com/conbench/conbench/assets/11835246/17e96bd3-0846-470d-9b5c-20474139f793">

But to share it to another Workspace (like pandas), it seems that I have to:
1) Add a Redirect URLs: 
But I can't get it right. I think that the scope is channels:read and chat:write, but I'm entering something wrong as I can't save it. 
The example is as follows, but how do I add the scope (I tried so many combinations already):
```
<a href="https://slack.com/oauth/v2/authorize?scope=incoming-webhook&client_id=33336676.569200954261">Add to Slack</a>
```

2) Remove Hard Coded Information:
Do I have to do token rotation? I'm kind of lost here (too).
  ```
Before activating public distribution, please remove any hard-coded information from your app, such as [OAuth tokens](https://api.slack.com/apps/A06EFL451QQ/oauth) and [webhook URLs](https://api.slack.com/apps/A06EFL451QQ/incoming-webhooks). Going forward, you should authorize your app for use on your workspace with [OAuth](https://api.slack.com/docs/oauth), just like anybody installing your app on their own workspace would do.
```

Could you please tell me if I am on the right track?

BTW, I'm working on sending conbench alerts by email, without going through gitHub. It works! Maybe I can share it in the future.. ?

**Comment by austin3dickey** [comment]
Hey, I'm sorry to say I haven't experimented with moving Slack apps between workspaces. My app is only for my own workspace. All I can tell you is that `benchalerts` uses the `chat.postMessage` endpoint, so you can use this for a minimal way to test your setup without having to set up `benchalerts`. If the following works (substituting ABCDEF with a real channel ID), then `benchalerts` should work too.
```
curl \
    -H 'Content-Type: application/json' \
    -H "Authorization: Bearer $SLACK_TOKEN" \
    -d '{"channel": "ABCDEF", "text": "test"}' \
    https://slack.com/api/chat.postMessage
```

**Comment by DeaMariaLeon** [comment]
Thanks. 

I wasn't able to use either conbench's Slack or its gitHub alert. There are too many false regressions detected. 
That's because the server where the asv benchmarks are run, has a lot of noise. Not conbench's fault. 

So I added to my code a way to "clean up" conbench's report. If it detected the regression more than 4 times (this threshold can be changed), then it's taken as a real regression. Then I receive the cleaned up report by email. 
With that, I open the issues to pandas' repo. 

This is just a "user feedback" - **not** a complain. And I wonder why you don't have the same problem with Arrow's benchmarks.

**Comment by DeaMariaLeon** [comment]
Sorry, I forgot to say that besides not being able to use 100% the automatic features yet, it's working real well. A lot of regressions have been detected and fixed in pandas. I find that amazing.

**Comment by austin3dickey** [comment]
>I wonder why you don't have the same problem with Arrow's benchmarks.

Haha, unfortunately this is a bit of a problem in Arrow's benchmarks as well. Could you say a little more about how you implemented the persistence logic? How are you keeping track of the number of regressions on the client side? I've thought about implementing something like this on the server side (see https://github.com/conbench/conbench/issues/1249), though it would probably require some refactoring.

>A lot of regressions have been detected and fixed in pandas. 

This is so great! Thanks for trying out this project; I'm glad it's working well for you.

**Comment by DeaMariaLeon** [comment]
Benchmarks for pandas are run on merged commits only (already in main). That means I can check what happens _after_ a regressions is found. When a regression is detected, I check that the next 4 commits fail the same benchmark. (The "4" is a threshold value I can change). 

I do that by manipulating a dataframe I build after running the alert pipeline. I only use the `['GetConbenchZComparisonStep'].results_with_z_regressions` of each commit.  

I use pandas' "rolling window" (to find the 4 commits in a row with the same regression). Then I also have to find the first commit with the alert. 

In short, it's just data cleaning. I'm not using all the cool classes that you developed. :-)  If you need more details let me know please.

>Thanks for trying out this project
Thank you!
